test/functional/test_framework/wallet.py

```diff
@@ -214,16 +214,19 @@ def get_address(self):
         assert_equal(self._mode, MiniWalletMode.ADDRESS_OP_TRUE)
         return self._address
 
-    def get_utxo(self, *, txid: str = '', vout: Optional[int] = None, mark_as_spent=True) -> dict:
+    def get_utxo(self, *, txid: str = '', vout: Optional[int] = None, mark_as_spent=True, avoid_immature_coinbase=True) -> dict:
         """
         Returns a utxo and marks it as spent (pops it from the internal list)
 
         Args:
         txid: get the first utxo we find from a specific transaction
         """
         self._utxos = sorted(self._utxos, key=lambda k: (k['value'], -k['height']))  # Put the largest utxo last
+        mature_coins = list(filter(lambda utxo: not utxo['coinbase'] or COINBASE_MATURITY <= utxo['confirmations'], self._utxos))
```

Comment:
In 894b2531991f8179022c158c2c15c06f0b774f0e "test: fix intermittent issue in `feature_bip68_sequence`"

The confirmation counts for the UTXOs can be very outdated when blocks are generated by things other than the MiniWallet. It would be nice if these could be kept up to date so that this filter is accurate.

Reply:
Since we already have the height of a UTXO, this could instead get the current block height and compute the confirmation count on the fly rather than using the stored confirmations. In fact, we probably shouldn't have the confirmation count for each UTXO at all since it can become outdated very quickly.

Reply:
> The confirmation counts for the UTXOs can be very outdated when blocks are generated by things other than the MiniWallet. It would be nice if these could be kept up to date so that this filter is accurate.

So, can I update `get_utxos` as well?